pkg/gce-pd-csi-driver/controller.go

```diff
@@ -458,3 +519,52 @@ func diskIsAttachedAndCompatible(volume *compute.Disk, instance *compute.Instanc
 	}
 	return false, nil
 }
+
+func pickTopology(top *csi.TopologyRequirement) (string, error) {
+	reqTop := top.GetRequisite()
+	prefTop := top.GetPreferred()
+
+	// Pick the preferred topology in order
+	if len(prefTop) != 0 {
+		if prefTop[0].GetSegments() == nil {
```

Comment:
You could have a common function to pick extract a zone from a topology, to simplify the preference vs required logic.

Reply:
picking a zone from topology is different between preference vs required so it doesn't make sense to me to abstract this out, I may be misunderstanding your suggestion